main/src/modules/contextualLayer/planner/src/planner.cpp

```diff
@@ -0,0 +1,593 @@
+    }
+    
+    grpPlans = rf.findGroup("PLANS");
+    avaiPlansList = *grpPlans.find("plans").asList();
+
+    bool ears = 1;
+    bool homeo = 1;
+    bool SM = 1;
+    bool BM = 1;
+
+    rpc.open(("/" + moduleName + "/rpc").c_str());
+    attach(rpc);
+
+    if (ears)
+    {
+        while (!Network::connect("/ears/target:o",rpc.getName())) {
+            yWarning() << "ears is unreachable";
+            yarp::os::Time::delay(0.8);
+        }
+    }
+
+    if (BM)
+    {
+        portToBehavior.open("/" + moduleName + "/behavior/cmd:o");
+        while (!Network::connect(portToBehavior.getName(),"/BehaviorManager/trigger:i")) {
+            yWarning() << "BehaviorManager is unreachable";
+            yarp::os::Time::delay(0.8);
+        }
+        yDebug()<<"Connected to BM!";
+    }
+
+    if (homeo)
+    {
+        toHomeo.open("/manager/toHomeostasis/rpc:o");
+        while (!Network::connect(toHomeo.getName(),"/homeostasis/rpc")) {
+            yWarning() << "homeostasis is unreachable";
+            yarp::os::Time::delay(0.8);
+        }
+    }
+
+    if (SM)
+    {
+        getState.open("/planner/state:i");
+        while (!Network::connect(getState.getName(), "/SensationManager/rpc")) {
+            yWarning() << "sensationManager is unreachable.";
+            yarp::os::Time::delay(0.8);
+        }
+    }
+
+    actPt = action_list.begin();
+    prioPt = priority_list.begin();
+
+    priority_list.clear();
+    action_list.clear();
+    id=0;
+    fulfill=0;
+    attemptCnt = 0;
+
+    yInfo() << "\n \n" << "----------------------------------------------" << "\n \n" << moduleName << " ready ! \n \n ";
+    
+    return true;
+}
+
+
+bool Planner::close() {
+    if(iCub) {
+        iCub->close();
+        delete iCub;
+    }
+
+    portToBehavior.interrupt();
+    portToBehavior.close();
+
+    port_behavior_context.interrupt();
+    port_behavior_context.close();
+
+    rpc.interrupt();
+    rpc.close();
+
+    return true;
+}
+
+bool Planner::freeze_all()
+{
+    // Prepare command
+    gandalf.clear();
+    gandalf.addString("freeze");
+    gandalf.addString("all");
+    // Send command
+    if (!Network::isConnected(toHomeo.getName(),"/homeostasis/rpc"))
+    {
+       yInfo() << Network::connect(toHomeo.getName(),"/homeostasis/rpc");
+       yarp::os::Time::delay(0.1);
+    }
+    toHomeo.write(gandalf);
+    yInfo() << "Gandalf has spoken.";
+
+    return true;
+}
+
+bool Planner::unfreeze_all()
+{
+    // Prepare command
+    gandalf.clear();
+    gandalf.addString("unfreeze");
+    gandalf.addString("all");
+    // Send command
+    if (!Network::isConnected(toHomeo.getName(),"/homeostasis/rpc"))
+    {
+        yInfo() << Network::connect(toHomeo.getName(),"/homeostasis/rpc");
+        yarp::os::Time::delay(0.1);
+    }
+    toHomeo.write(gandalf);
+    yInfo() << "Gandalf has rescinded.";
+
+    return true;
+}
+
+bool Planner::checkKnown(const Bottle& command, Bottle& avaiPlansList, string foundPlan) {
```

Comment:
`foundPlan` is useless. It is nowhere used. If you want to print the plan below, just print `yInfo() << "plan is known: " + avaiPlansList.get(i).asString()`